core/src/main/scala/cilib/Entity.scala

```diff
@@ -19,5 +19,93 @@ object Entity {
       def fitness(a: Entity[S, A]) =
         a.pos.objective
     }
+}
+
+object BCHM {
+  def clamp[S, A: scalaz.Equal](x: Entity[S, A])(implicit N: spire.math.Numeric[A]) =
+    Lenses._position.modify((p: Position[A]) => {
+      val c: NonEmptyList[A] = p.pos.zip(p.boundary).map {
+        case (p, b) => {
```

Comment:
You can lose the curly braces here, seeing as the expression is only an `if` and always has a resulting value. This if based version is much better imho!

What I means is:

```scala
   case (p, b) =>
      if ....
```